plugins/catalog-backend/src/catalog/CatalogLogic.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger } from 'winston';
+import { LocationReader } from '../ingestion/types';
+import { Catalog } from './types';
+
+export class CatalogLogic {
+  public static startRefreshLoop(
+    catalog: Catalog,
+    reader: LocationReader,
+    logger: Logger,
+  ): () => void {
+    let cancelled = false;
```

Comment:
Another pattern for this would a cancellation promise that cancels the task when resolved. That way you can use `Promise.race` to end earlier. Would probably only use that if we want to wait for shutdown though.

Reply:
mm not sure if worth, could look at that tomorrow morning

Reply:
ye no I don't think it's worth it :grin:

Reply:
mm did it anyway (where's my Deferred??)